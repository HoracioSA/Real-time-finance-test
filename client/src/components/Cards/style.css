.cardContainer{
    display: flex;
    justify-content:space-between;
    align-items:center;
    width:13rem;
    height: 5em;
    background:#fff;
    box-shadow: 0px 2px 16px rgba(32, 108, 164, 0.16);
    border-radius: 8px;
    padding:0.7rem;
    margin-right: 0.7rem;
    margin-top: 3rem;
    cursor: pointer;
}
.iconAndTitleContainer{
    display: flex;
    gap: 1.5rem;
    align-items:center;
}
.iconAndTitleContainer button{
    width: 2.5rem;
    border: 0;
    height: 2.5rem;
    border-radius: 50px;
    background:var(--rose-gradient);
    justify-content:center;
    align-content:center;
    align-items: center;
}
.iconAndTitleContainer button >img{
    width:1.7rem;
    height: 1.7rem;
}
.price_company{
    margin-left: 1rem;
}

