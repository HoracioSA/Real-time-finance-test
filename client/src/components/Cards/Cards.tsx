
import { FiArrowUp } from 'react-icons/fi'
import './style.css'

export interface Ticker {
    ticker?:string;
    exchange: string;
    price: string;
    change: string;
    change_percent:string;
    dividend?: string;
    last_trade_time?:Date;
    yields?:string;
}

const Cards =({ticker,exchange,price,change,
    change_percent,
    dividend,
    last_trade_time,
    yields}:Ticker) =>{
    
    return (
        <div className="cardContainer">
            <div className="iconAndTitleContainer">
                <button>
                    <FiArrowUp size={17} color="#C774EE" />
                </button>
                <div className="price_company">
                    <strong>{exchange}</strong>
                    <p>{price}</p>
                </div>
            </div>
            
                <div className="wining_percent">
                    <strong><span>+</span>{change}</strong>
                    <p><span>-</span>{change_percent}</p>
                </div>
        </div>

    )
}
export default Cards
