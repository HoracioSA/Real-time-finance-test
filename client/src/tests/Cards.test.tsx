import { getByTestId, render, screen } from '@testing-library/react';
import Cards from '../components/Cards/Cards';


test('<Cards> learn react link', () => {
 const component =render(<Cards key="ticker"
    exchange="NASDAQ"
    price="300"
    change="0.9"
    change_percent="0.5"
    dividend="7"
    />);
  expect(component.container).toMatchSnapshot()
});

