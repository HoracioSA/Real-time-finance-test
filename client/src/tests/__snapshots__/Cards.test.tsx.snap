// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cards> learn react link 1`] = `
<div>
  <div
    class="cardContainer"
  >
    <div
      class="iconAndTitleContainer"
    >
      <button>
        <svg
          color="#C774EE"
          fill="none"
          height="17"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          style="color: rgb(199, 116, 238);"
          viewBox="0 0 24 24"
          width="17"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="12"
            x2="12"
            y1="19"
            y2="5"
          />
          <polyline
            points="5 12 12 5 19 12"
          />
        </svg>
      </button>
      <div
        class="price_company"
      >
        <strong>
          NASDAQ
        </strong>
        <p>
          300
        </p>
      </div>
    </div>
    <div
      class="wining_percent"
    >
      <strong>
        <span>
          +
        </span>
        0.9
      </strong>
      <p>
        <span>
          -
        </span>
        0.5
      </p>
    </div>
  </div>
</div>
`;
